app/models/ocelot/stanzas/CalculationStanza.scala

```diff
@@ -41,4 +42,91 @@ object CalculationStanza {
         (JsPath \ "next").write[Seq[String]] and
         (JsPath \ "stack").write[Boolean]
       )(unlift(CalculationStanza.unapply))
+}
+
+sealed trait Operation {
+
+  val logger: Logger = Logger(this.getClass)
+
+  val left: String
+  val right: String
+  val label: String
+
+  def eval(labels: Labels) : Labels
+
+  def value(arg: String, labels: Labels): String = labelReference(arg).fold(arg){ref => labels.value(ref).getOrElse("")}
+
+  def addString(s1: String, s2: String) : Option[String] = Option(s1 + s2)
+
+  def subtractString(s1: String, s2: String) : Option[String] = {
```

Comment:
This could be unsupportedOperation() and re-used elsewhere as it is likely, that most, if not all of the remaining calculation ops will be unsupported on non numeric data